OpenRA.Mods.Common/Orders/PlaceBuildingOrderGenerator.cs

```diff
@@ -210,5 +219,25 @@ public IEnumerable<IRenderable> RenderAfterWorld(WorldRenderer wr, World world)
 		}
 
 		public string GetCursor(World world, CPos xy, MouseInput mi) { return "default"; }
+
+		IEnumerable<Order> TryClearBlockers(World world, CPos topLeft)
+		{
+			var allTiles = FootprintUtils.Tiles(world.Map.Rules, building, buildingInfo, topLeft).ToArray();
+			var neightborTiles = Util.ExpandFootprint(allTiles, true).Except(allTiles)
+				.Where(t => world.Map.Contains(t)).ToList();
+
+			var blockers = allTiles.SelectMany(t => world.ActorMap.GetUnitsAt(t))
+				.Where(a => a.Owner == producer.Owner && a.HasTrait<Mobile>());
```

Comment:
I started with `IMove`, but `Mobile` gives me the `.CanEnterCell(t)` that I use below, so I figured this won't affect aircraft anyway, might as well use `Mobile`.